### ISSUE TITLE
[BUG] Clicking on/Click dragging the root scrollbar closes SfSidebar

### ISSUE THREAD

**Issue by kingrea** [vuestorefront/storefront-ui#2478]
**Describe the bug**
If you have an SfSidebar open on a page, then attempt to use the pages scrollbar (not a scrollbar on SfSidebar) - it will close the sidebar. 

**How to reproduce**
1. Go to any website that uses vue storefront UI and content high enough that creates a verticle scroll
2. Click on an element that opens a sidebar with non-persistence 
3. Click, or click and drag the scrollbar
4. The sidebar will close

or if you want to look at it in action of the docs:

1. https://docs.storefrontui.io/?path=/docs/components-organisms-sidebar--common
2. (in chrome) right click the sidebar (not the overlay) and inspect element
3. find the div with id="root" and add 'overflow: scroll, height: 1000px'
4. Attempt to use the root elements scrollbar

**Expected behavior**
The menu should not close

**Actual behavior**
The menu closes

**Devices (please complete the following information):**
 - Device: [Desktop]
 - OS: [Windows]
 - Browser: [chrome]

**Additional context**
The solution is probably close to what I found in [this post](https://github.com/kentor/react-click-outside/issues/9). 

It's an edge case. Scrollbars are technically outside of the clients width. However since the clickOutside directive (utilities/directives/click-outside/click-outside-directive.js) doesn't check to see if the click is happening outside of the client width - this is expected behaviour. 
It needs something like:
`if (e.offsetX <= e.target.clientWidth) { onClickOutsideEvent(e); }
`
to check if the click event is outside of the documents scope.

**Comment by AdamPawlinski** [comment]
Hi @kingrea, thanks for this issue. Would you like to take this task?

**Comment by shatanov** [comment]
Hi @AdamPawlinski, maybe I can take this task?

**Comment by shatanov** [comment]
> Hi @AdamPawlinski, maybe I can take this task?

@FRSgit

**Comment by Szymon-dziewonski** [comment]
Hey @shatanov feel free to take this task, it will be highly welcomed :)

**Comment by shatanov** [comment]
> Hey @shatanov feel free to take this task, it will be highly welcomed :)

Ok) pls, assign this task to me

**Comment by shatanov** [comment]
> > Hey @shatanov feel free to take this task, it will be highly welcomed :)
> 
> Ok) @Szymon-dziewonski  pls, assign this task to me

**Comment by Szymon-dziewonski** [comment]
@shatanov you are assigned, have fun and good luck :)

**Comment by FRSgit** [comment]
If you have any questions feel free to reach us on [discord](https://discord.com/invite/vuestorefront)!

**Comment by shatanov** [comment]
> If you have any questions feel free to reach us on [discord](https://discord.com/invite/vuestorefront)!

Yep, I have on discord server, but I haven`t conributor status there
